src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs

```diff
@@ -0,0 +1,225 @@
+
+        /// <summary>
+        /// The relative directory path of the current node from the root node.
+        /// </summary>
+        public string Path { get; private set; } = "";
+
+        /// <summary>
+        /// Dictionary of labels and Path Item objects that describe the operations available on a node.
+        /// </summary>
+        public IDictionary<string, OpenApiPathItem> PathItems { get; } = new Dictionary<string, OpenApiPathItem>();
+
+        /// <summary>
+        /// A dictionary of key value pairs that contain information about a node.
+        /// </summary>
+        public IDictionary<string, List<string>> AdditionalData { get; set; } = new Dictionary<string, List<string>>();
+
+        /// <summary>
+        /// Flag indicating whether a node segment is a path parameter.
+        /// </summary>
+        public bool IsParameter => Segment.StartsWith("{");
+
+        /// <summary>
+        /// The subdirectory of a relative path.
+        /// </summary>
+        public string Segment { get; private set; }
+
+        /// <summary>
+        /// Flag indicating whether the node's PathItems has operations.
+        /// </summary>
+        /// <returns>true or false.</returns>
+        public bool HasOperations(string label)
+        {
+            if ((bool)!PathItems?.ContainsKey(label))
+            {
+                return false;
+            }
+
+            return PathItems[label].Operations != null && PathItems[label].Operations.Count > 0;
+        }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="segment">The subdirectory of a relative path.</param>
+        private OpenApiUrlTreeNode(string segment)
+        {
+            Segment = segment;
+        }
+
+        /// <summary>
+        /// Creates an empty structured directory of <see cref="OpenApiUrlTreeNode"/> node.
+        /// </summary>
+        /// <returns>The root node of the created <see cref="OpenApiUrlTreeNode"/> directory structure.</returns>
+        public static OpenApiUrlTreeNode Create()
+        {
+            return new OpenApiUrlTreeNode("/");
+        }
+
+        /// <summary>
+        /// Creates a structured directory of <see cref="OpenApiUrlTreeNode"/> nodes from the paths of an OpenAPI document.
+        /// </summary>
+        /// <param name="doc">The OpenAPI document.</param>
+        /// <param name="label">Name tag for labelling the <see cref="OpenApiUrlTreeNode"/> nodes in the directory structure.</param>
+        /// <returns>The root node of the created <see cref="OpenApiUrlTreeNode"/> directory structure.</returns>
+        public static OpenApiUrlTreeNode Create(OpenApiDocument doc, string label)
+        {
+            Utils.CheckArgumentNull(doc, nameof(doc));
+            Utils.CheckArgumentNullOrEmpty(label, nameof(label));
+
+            OpenApiUrlTreeNode root = new OpenApiUrlTreeNode("/");
+
```

Comment:
in [our copy](https://github.com/microsoft/kiota/blob/b8c8c87e71d8a9db38b2a5bfc20d894350b748d5/src/Kiota.Builder/OpenApiUriSpaceNode.cs#L65) we init with empty string. I'm not suggesting any change here, I'm just curious about the reason behind the difference?

Reply:
I figured that the root path segment should ideally be a `/` instead of just an empty string. In the Graph OpenAPI docs, at least, we represent it as such.

Reply:
In [the powershell open api descriptions](https://github.com/microsoftgraph/msgraph-sdk-powershell/blob/dev/openApiDocs/v1.0/Mail.yml#L0-L1) the servers already contain a trailing slash. I'm not sure whether this is just us or a standard practice in the community. What I'm wondering is whether starting with a slash here will lead to a double slash and/or to consumers of the open api lib to have to control for that?
The paths also always start with a slash.

If we look at [pet store](https://petstore.swagger.io/v2/swagger.json) which is v2, they use host + base path with no trailing slash, but the paths also all start with a slash.